scripts/03-site/hooks/samples/post-module-load.sh

```diff
@@ -123,115 +82,33 @@ analytics_lmod_send_version() {
   amplitude_lmod_send_event "${mod_name}" "${mod_ver}" "$2" "${mod_arch}"
 }
 
-amplitude_lmod_prepare_event() {
-  local mod_name="${1:-}"
-  local mod_version="${2:-}"
-  local mod_version_long="${3:-}"
-  local mod_arch="${4:-}"
-
+amplitude_lmod_send_event() {
+  local mod_name="${1:-}" mod_version="${2:-}" mod_version_long="${3:-}" mod_arch="${4:-}"
   if [ -z "${mod_name}" ] || [ -z "${mod_version}" ] || [ -z "${mod_version_long}" ]; then
     [ -n "${SPACK_HOOK_DEBUG:-}" ] && echo "Skipping Amplitude event: missing module details" >&2
-    return 1
+    return 0
   fi
 
-  local username="${USER:-$(whoami 2>/dev/null || echo)}"
-  local hostname_fqdn="$(hostname -f 2>/dev/null || hostname 2>/dev/null || echo)"
-  local session_date="$(date +%Y-%m-%d)"
-  local device_id="${username}@${_amplitude_cluster_id}/${session_date}"
-  local user_id="${username}@${_amplitude_cluster_id}"
-
-  local _amplitude_spack_variant="${SPACK_VARIANT:-unknown}"
-  local _amplitude_spack_disable_local_config="${SPACK_DISABLE_LOCAL_CONFIG:-0}"
-  local _amplitude_spack_root="${SPACK_ROOT:-unknown}"
-  local _amplitude_spack_user_cache_path="${SPACK_USER_CACHE_PATH:-unknown}"
-  local _amplitude_spack_user_config_path="${SPACK_USER_CONFIG_PATH:-unknown}"
-
-  # Optional extra event properties
-  local extra_props=""
-  add_json_prop() {
-    if [ -n "$2" ]; then
-      extra_props+=$(printf ', "%s": "%s"' "$1" "$(json_escape "$2")")
-    fi
-  }
-  add_json_prop "module_architecture" "${mod_arch}"
-  unset -f add_json_prop
-
-  # Build JSON payload (single event)
-  local json_data
-  json_data=$(printf '{
-  "api_key": "%s",
-  "events": [{
-    "device_id": "%s",
-    "user_id": "%s",
-    "event_type": "Module Load",
-    "event_properties": {
-      "cluster": "%s",
-      "username": "%s",
-      "hostname": "%s",
-      "module_name": "%s",
-      "module_version": "%s",
-      "module_version_long": "%s",
-      "spack_variant": "%s",
-      "spack_disable_local_config": "%s",
-      "spack_root": "%s",
-      "spack_user_cache_path": "%s",
-      "spack_user_config_path": "%s"%s
-    },
-    "user_properties": {
-      "cluster": "%s",
-      "username": "%s"
-    }
-  }]
-}' \
-    "$(json_escape "${_amplitude_api_key}")" \
-    "$(json_escape "${device_id}")" \
-    "$(json_escape "${user_id}")" \
+  local event_props user_props
+  event_props=$(printf '{"cluster":"%s","username":"%s","hostname":"%s","module_name":"%s","module_version":"%s","module_version_long":"%s","spack_variant":"%s","spack_disable_local_config":"%s","spack_root":"%s","spack_user_cache_path":"%s","spack_user_config_path":"%s"%s}' \
```

Comment:
can this whole part refactor to common, will send basic props in any case

Reply:
eg basic prop + read list of key1 value1 key2 value2, key must start with slurm|spack|module_